test/tests/functional/pbs_unconfirm_resv_end_hook.py

```diff
@@ -0,0 +1,67 @@
+from tests.functional import *
+import time
+
+
+class TestUnconfirmedReservation(TestFunctional):
+    """
+    Testsuite to verify the reservation end hook running for a confirmed reservation.
+    """
+
+    def setUp(self):
+        """ Create and import the python script for reservation end hook
+        in setup method"""
+
+        super(TestUnconfirmedReservation, self).setUp()
+        hook_script = """
+import pbs
+e=pbs.event()
+
+pbs.logmsg(pbs.LOG_DEBUG, 'Reservation End Hook name - %s' % (pbs.event().hook_name))
+
+if e.type == pbs.RESV_END:
+    pbs.logmsg(pbs.LOG_DEBUG, 'Reservation ID - %s'%pbs.event().resv.resvid)"""
+
+        self.hook_name = 'resvend_hook'
+        attrs = {'event': 'resv_end'}
+        self.server.create_import_hook(
+            self.hook_name, attrs, hook_script, overwrite=True)
+        self.server.manager(
+            MGR_CMD_SET, SERVER, {
+                'log_events': 2047}, expect=True)
+
+    def submit_resv(self, offset, duration, select="1:ncpus=1"):
+        """ Submit the advance reservation """
+
+        start_time = time.time() + offset
+        end_time = start_time + duration
+        attrs = {
+            'reserve_start': start_time,
+            'reserve_end': end_time,
+            'Resource_List.select': select}
+        rid = self.server.submit(Reservation(TEST_USER, attrs))
+
+        return rid
+
+    def test_unconfirmed_resv(self):
+        """ Test to verify the resvend hook does not run for an unconfirmed reservation.
+            Can be verified when it is single node setup """
```

Comment:
Could you rephrase this some?  Instead of saying 'an unconfirmed reservation', say 'for a reservation that does not get confirmed'.  You might also consider adding a test when you delete a reservation that is in the unconfirmed state.  You do this by taking the scheduler down before you submit your reservation.  The reservation will stay unconfirmed.  You can then delete it.

Reply:
@bhroam Thank you for the review. The test case sounds reasonable to be. I will update the same.